Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -152,24 +169,31 @@ private void GenerateCorrelationIdAndAddToDictionary(string ikey, string appId)
         {
             try
             {
+#if !NETCORE
                 SdkInternalOperationsMonitor.Enter();
-
+#endif
                 Uri appIdEndpoint = this.GetAppIdEndPointUri(instrumentationKey);
-
+#if !NETCORE
                 WebRequest request = WebRequest.Create(appIdEndpoint);
                 request.Method = "GET";
 
                 using (HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync().ConfigureAwait(false))
+                using (StreamReader reader = new StreamReader(response.GetResponseStream()))
                 {
-                    using (StreamReader reader = new StreamReader(response.GetResponseStream()))
-                    {
-                        return await reader.ReadToEndAsync();
-                    }
+                    return await reader.ReadToEndAsync();
+                }
+#else
+                using (HttpClient client = new HttpClient())
+                {
+                    return await client.GetStringAsync(appIdEndpoint);
```

Comment:
.ConfigureAwait(false). Probably not very important with 0 wait time for first call (as things are today) - but if the timout in the caller was non zero - you would want to get to that timer which this async call was happening. Hence encouraging the framework to queue continuation on a separate thread is a good idea.